src/screens/Reader.tsx

```diff
@@ -173,21 +174,29 @@ export default function Reader() {
             formData.append('read', 'true');
             client.patch(`/api/v1/manga/${manga.id}/chapter/${chapter.index}`, formData);
 
-            history.replace({
-                pathname: `/manga/${manga.id}/chapter/${chapter.index + 1}`,
-                state: history.location.state,
+            openNextChapter(chapter as IChapter, settings.skipDupChapters, (nextChapterIndex) =>
+                history.replace({
+                    pathname: `/manga/${manga.id}/chapter/${nextChapterIndex}`,
+                    state: history.location.state,
+                }),
+            ).catch(() => {
+                // openNextChapter shows error toast
```

Comment:
I'd like to keep it that way to reduce duplication.
the default error handling is showing the toast and in case some state has to be reset in can be done separately.

but overall I don't like the way it is done.
imo changing the chapter should be handled in one place and the others just call the function not having to care about handling the error at all, but thats currently not possible I think

Reply:
You mean prevent duplication with `ReaderNavBar`?

> but overall I don't like the way it is done.

It could be changed to be done correctly. Have chapter navigation handled in `Reader.tsx` and pass chapter change callbacks to `ReaderNavBar` via props. Then you would only have single place where navigation is handled and you could handle toasts and all there.

If this is staying then I have other question. Why is the `catch` here? When error is handled, it should not be rethrown from the inside. Then this should not need to handle errors.

Reply:
> If this is staying then I have other question. Why is the catch here? When error is handled, it should not be rethrown from the inside. Then this should not need to handle errors.

the toast is for informing the user, raising the error is so that the callee can handle the error.
in the reader nav bar I disable the chapter nav buttons so that it's not possible to click them while the next chapter is being retrieved, in case this fails, the buttons have to be enabled again.

> It could be changed to be done correctly. Have chapter navigation handled in Reader.tsx and pass chapter change callbacks to ReaderNavBar via props. Then you would only have single place where navigation is handled and you could handle toasts and all there.

if I'm not missing something, it is currently not really possible to enable the loading mode of the reader without having to change the manga object.
so it would still be necessary for the nav bar to disable/enable the chapter nav buttons, unless this is not wanted, then it can just be removed, and the error doesn't have to get raised

that is want I meant with I don't like how it's done, imo the reader has to be cleaned up to make adding changes easier and I'm not going to change that in this pr, since that has nothing to do with the change imo

Reply:
moved util functions to Reader and passing functions to open next and prev chapter to ReaderNavBar
not throwing error anymore and passing disable chapter nav buttons flag as prop

Reply:
https://github.com/Suwayomi/Tachidesk-WebUI/pull/268